README.md

```diff
@@ -1,36 +1,54 @@
+14. Click "Register an application"
+15. Fill out application details:
+- Name it appropriately (e.g. Onestop Integration - My Name)
+- Company and website URL don't matter
+- Select Basecamp 4 under Products
+- For OAuth2 redirect URI, fill in the URI with the following, and replace `YOUR_SCRIPT_ID` with your Google Apps Scripts ID from the earlier section: `https://script.google.com/macros/d/YOUR_SCRIPT_ID/usercallback`. 
+16. Click "Register this app"
+17. Open your app and copy your Client ID and Client Secret for the following step.
+19. Create a new file `environmentVariables.ts` inside the [config](./config/) folder, with the following contents:
+```js
+export const BASECAMP_CLIENT_ID = 'REPLACE_WITH_YOUR_BASECAMP_CLIENT_ID';
+export const BASECAMP_CLIENT_SECRET = 'REPLACE_WITH_YOUR_BASECAMP_CLIENT_SECRET';
 ```
 
-Create an Apps Script script, if you don't already have one you want to use.
-- You probably want a [container-bound script](https://developers.google.com/apps-script/guides/bound#access_to_bound_scripts) linked to the OneStop spreadsheet.
-
-Make your own copy of the OneStop here: https://drive.google.com/drive/folders/1s_u-hmstlLL1JqRKyKJ7r2AMyyTCCiut
-
-Get the Script ID for your personal dev OneStop by following these steps:
-- Open your new sheet
-- Click Extensions > App Script
-- Select "Create a new project"
-- Give it any name you want
-- Go to settings
-- The Script ID should be in the settings page
-- Copy the script id 
+## Development
+### Making changes
+Create a feature branch: `git checkout -b 'branch-feature-name'`, and name the branch with a clear and concise description of the change.
 
-Fill out [`.clasp.json`](./.clasp.json) with your created script ID.
+To allow functions to be run from the Google Apps Scripts, the function must have the `export` keyword. Additionally, it needs to be exported in [index.ts](./src/index.ts).
 
+### Testing changes
 ```shell
 npm run build
 npm run deploy
 ```
+This will deploy your code to your personal onestop sheet and app script, and use your personal Basecamp integration app.
 
+To run functions, make sure you first open your Google sheet (onestop copy), then open your script from Extensions > App Script. Keep the Google sheet open in a separate tab, as it will be necessary for any Spreadsheet UI interactions. In the App Script code editor, find your function from the dropdown and hit run. If your function makes use of the Spreadsheet UI API, you will need to click back to your Google sheet tab for the interaction.
 
+#### Testing Basecamp API's
+You will need an OAuth2 access token, which you can receive a link for when making any Basecamp request when unauthorized. The easiest way is to call the `checkAuthorization` function from the Google Apps Scripts editor, click back to your Google sheet tab, then follow the instruction in the alert dialog (open the auth link in a new tab, close it, then re-run the function).
```

Comment:
Not a blocking review comment, just a question. Does this access token need to be stored somewhere?

Reply:
it's automatically stored as part of the oauth2 library, i believe in property service. looks like the access token usually lasts for a few weeks before it needs to be renewed